highway.tex

```diff
@@ -24,807 +26,1323 @@
 
+\subsection{Consensus in the Context of Blockchain}
 
-\subsection*{Overview}
+In a typical blockchain system, validators are tasked with performing an iterative consensus on an ever-growing chain of transactions that they receive from the external environment. 
+In the process, they enclose transactions into blocks forming a blockchain, in which each block refers to its predecessor by hash.
+The first one, the \emph{genesis block}, is part of the protocol definition. 
 
-In section \ref{sectionMessages} we introduce the basic concepts of messages, justifications and equivocations, and analyze the safety guarantees that can be derived from their structure so that we can determine in a general setting when consensus decisions are final, i.e. under some reasonable assumptions they cannot be reverted.
+In Highway, as the set of validators is either constant or subject only to very controlled changes (between different eras, see Subsection \ref{sec:eras}), specific validators are directly appointed to construct a block in a given time slot.
+They do so by enclosing transactions from their local queue and hash of the block that they believe should be the predecessor.
 
-Section \ref{sectionBlockchain} provides an explanation about how these results can be applied to the specific use case of a blockchain.
+As it may happen that validator does not refer to the last constructed block (either intentionally, or due to the network failure), the set $\mathcal{B}$ of blocks is a tree, with a unique path leading from each block to the root: the genesis block $G$. 
+The main goal of the consensus protocol in such a scenario is to choose a single branch from such a tree.
+For a block $B$, we refer to all the blocks that are on the other branches as \emph{competing} with $B$, as if any of them would be chosen, $B$ could not.
 
-In section \ref{sectionLiveness}, we introduce additional rules that guarantee not only safety, but also liveness; decisions can be relied on and are eventually actually made.
 
-Finally, section \ref{sectionPermissionless} puts the abstract consensus algorithm in the context of a distributed proof of stake network, outlines how deposits, rewards and penalties can be handled, and addresses the problem of long range attacks.
+\subsection{Practical challenges}
 
-\newpage
+{\bf Strong optimistic finality.} While since the initial definition of the partially synchronous model by Dwork, Lynch and Stockmeyer~\cite{DLS88} the vast body of research was created to optimize various parameters of protocols in this setting, most of it was written under the semi-formal assumption that the existence of more than $n/3$ dishonest nodes predates the existence of any provable guarantees for such protocols.
```

Comment:
"the vast body" → "a vast body"?
"predates" → "precedes"? (Not sure… need a native speaker… @xcthulhu?)

Reply:
"predates" is correct.

I don't think we should use the phrase "vast body" here.

Reply:
```suggestion
{\bf Strong optimistic finality.} Partially synchronous consensus protocols originate with Dwork, Lynch and Stockmeyer~\cite{DLS88}. Most subsequent research has been into optimizing various parameters of protocols in this setting. This work generally assumes, semi-formally, that less than $n/3$ nodes are dishonest. However most publications predate the existence of any provable guarantees for such protocols.
```

Reply:
As the proposition of @xcthulhu has different meaning, I'm leaving as it is with the only change of "the vast body" -> "a vast body". The meaning of the original version was that the papers semi-fomrally assume that nothing can be proven in the presence of more than $n/3$ dishonest nodes.